### ISSUE TITLE
[macOS] Setting the size of a window will crash 

### ISSUE THREAD

**Issue by JenniferLee520** [red/red#2381]
click the "bigger" button to see the crash
```Red
Red []

view win: make face! [
    type: 'window
    offset: 50x50
    size: 100x100
    pane: reduce [
        make face! [
            type: 'button
            offset: 0x0
            size: 100x30
            text: "bigger"
            actors: object [
                on-down: func [ face [object!] event [event!] ] [
                    face/parent/size: face/parent/size + 100x100
                ]                
            ]
        ]
    ]
]
```

**Comment by endo64** [comment]
What platform you are on?
It works as expected on Win10 x64.

**Comment by qtxie** [comment]
@endo64 It's on macOS.

**Comment by qtxie** [comment]
Implemented in commit:  d197e05
